.github/workflows/clang-format.yml

```diff
@@ -30,12 +30,21 @@ jobs:
       - name: Get the base and head branches
         id: fetchstep
         run: |
-          echo "BASE_BRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
-          echo "HEAD_BRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
```

Comment:
I don't think it is with the github recommended approach:
https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable

Reply:
It is good to have env variables but when I used BASE_BRANCH as env var, it didnt work for me citing that this can't be an env variable
${{ github.event.pull_request.base.ref }}
So, I used quotes on local variables as the fix for script injection

Reply:
Turns out BASE_BRANCH:${{ github.event.pull_request.base.ref }} is different from
BASE_BRANCH: ${{ github.event.pull_request.base.ref }} where the spacing matters. Going to try with the env variable ,if check works then all good now